DQMOffline/Trigger/plugins/MuonMonitor.cc

```diff
@@ -0,0 +1,288 @@
+
+  me.numerator   = ibooker.book2D(histname+"_numerator",   histtitle+" (numerator)",   nbinsX, arrX, nbinsY, arrY);
+  me.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, arrX, nbinsY, arrY);
+}
+
+void MuonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+
+  histname = "muon_pt"; histtitle = "muon PT";
+  bookME(ibooker,muonME_,histname,histtitle,muon_binning_.nbins,muon_binning_.xmin, muon_binning_.xmax);
+  setTitle(muonME_,"Muon pT [GeV]","events / [GeV]");
+
+  histname = "muon_pt_variable"; histtitle = "muon PT";
+  bookME(ibooker,muonME_variableBinning_,histname,histtitle,muon_variable_binning_);
+  setTitle(muonME_variableBinning_,"Muon pT [GeV]","events / [GeV]");
+
+  histname = "muonVsLS"; histtitle = "muon pt vs LS";
+  bookME(ibooker,muonVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,muon_binning_.xmin, muon_binning_.xmax);
+  setTitle(muonVsLS_,"LS","Muon pT [GeV]");
+
+  histname = "muon_phi"; histtitle = "Muon phi";
+  bookME(ibooker,muonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+  setTitle(muonPhiME_,"Muon #phi","events / 0.1 rad");
+
+
+  histname = "muon_eta"; histtitle = "Muon eta";
+  bookME(ibooker,muonEtaME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin,eta_binning_.xmax);
+  setTitle(muonEtaME_,"Muon #eta","events");
+
+
+  histname = "muon_etaphi"; histtitle = "Muon eta-phi"; 
+  bookME(ibooker,muonEtaPhiME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+  setTitle(muonEtaPhiME_,"#eta","#phi"); 
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void MuonMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+  
+  //float met = pfmet.pt();
+  //  float phi = pfmet.phi();
+
+
+  edm::Handle<reco::MuonCollection> muonHandle;
```

Comment:
ok I confirmed with Swagata and she told me for scouting trigger, they would like to see efficiency as a function of hlt object pt. I will add that information.